azurerm/resource_arm_media_services_test.go

```diff
@@ -0,0 +1,252 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/mediaservices/mgmt/2018-07-01/media"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMMediaServicesAccount_singlestorage(t *testing.T) {
+
+	ri := tf.AccRandTimeInt()
+	resourceName := fmt.Sprintf("%d", ri)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMMediaServicesAccountDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: generateTemplate_singlestorage(resourceName, testLocation()),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					checkAccAzureRMMediaServicesAccount("azurerm_media_services.ams", testLocation(), "1"),
+				),
+			},
+		},
+	})
+}
+
+func generateTemplate_singlestorage(name string, location string) string {
+	return fmt.Sprintf(`
+	resource "azurerm_resource_group" "testrg" {
+			name     = "%[1]s"
+			location = "%[2]s"
+	  }
+	  
+	  resource "azurerm_storage_account" "testsa" {
+			name                     = "%[1]s"
+			resource_group_name      = "${azurerm_resource_group.testrg.name}"
+			location                 = "${azurerm_resource_group.testrg.location}"
+			account_tier             = "Standard"
+			account_replication_type = "GRS"
+		
+			tags {
+				environment = "staging"
+			}
+	  }
+	  
+	  resource "azurerm_media_services" "ams" {
+	  
+			  name					= "%[1]s"
+			  location				= "${azurerm_resource_group.testrg.location}"
+			  resource_group_name 	= "${azurerm_resource_group.testrg.name}"
+			  
+			  storage_account {
+					  id			= "${azurerm_storage_account.testsa.id}"
+					  is_primary	= true
+			  }
+
+			  tags {
+				environment = "staging"
+			  }
+	  }
+`, name, location)
+}
+
+func TestAccAzureRMMediaServicesAccount_multiplestorage(t *testing.T) {
+
+	ri := tf.AccRandTimeInt()
+	resourceName := fmt.Sprintf("%d", ri)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMMediaServicesAccountDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: generateTemplate_multiplestorage(resourceName, testLocation()),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					checkAccAzureRMMediaServicesAccount("azurerm_media_services.ams", testLocation(), "2"),
```

Comment:
can we remove this function and instead check the values persisted in the state? there's an example of this here https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_dev_test_windows_virtual_machine_test.go#L27

Reply:
checkAccAzureRMMediaServicesAccount uses functions, such as resource.TestCheckResourceAttr internally to check the state. It only exists to remove the duplication since both acceptance tests have the same validation logic with slightly different values (ex: # of storage accounts).